### PR TITLE
Consul 1.0.0 compatibility

### LINKED ISSUE #130: Consul 1.0.0 compatibility
Consul 1.0.0 has some breaking API changes:
https://www.consul.io/docs/upgrade-specific.html#consul-1-0

One of them is that com.ecwid.consul.v1.agent.model.Self.Config must be broken down into two classes at least: Config and DebugConfig

### PR DESCRIPTION
**Pull request by lazystone**
Fixes #130 

The most important changes are made in AgentConsulClient - some requests are PUT now instead of GET.

Changes in build.gradle unimportant apart from embedded-consul dependency which supports Consul 1.0.0

I tried to split Config to Config and DebugConfig but to be honest, I just don't want to add tons of debug parameters, so I just left one that existed in old Config before and removed which does not exist in the new DebugConfig.

### CODE REVIEW

**Comment by vgv** [comment]
Hmm, I see the breaking changes in API.
I think it's time to bump consul-api version to 2.X.X

**Comment by lazystone** [comment]
@vgv changes in REST verbs probably are not breaking - seems like before Consul 1.0.0 didn't pay any attention to the verb(both GET and PUT work).
Config response is different though in Consul 1.0.0.

I would suggest to leave major version of API as it is - 1.x.x since event to keep it aligned with Consul. After all even HashiCorp didn't bump up their version number.

File: src/main/java/com/ecwid/consul/v1/agent/model/Self.java [hunk 1]
```diff
@@ -20,57 +20,106 @@
...
+							", nodeName='" + nodeName + "'" +
+							", revision='" + revision + '\'' +
+							", server=" + server +
+							", version='" + version + '\'' +
+							'}';
+		}
+
+	}
+
+	public static class DebugConfig {
```
Comment (preetapan) [inline src/main/java/com/ecwid/consul/v1/agent/model/Self.java]: like @slackpad suggested, we recommend not parsing DebugConfig. Its really an internal representation and we will add or remove fields in the future.

**Comment by lazystone** [comment]
@vgv Could you have a look if I should do anything more in this PR?

**Comment by preetapan** [comment]
@vgv would be very helpful if this PR is reviewed and merged and a new version of consul-api is published, given that spring-cloud also needs this. We would really appreciate it!

**Comment by vgv** [comment]
@lazystone 

Hi, thank you for RP. 
I will not merge this PR, however, I just take parts of your code and apply it by hands. 

Reasons:
1) AgentConsulClient already uses correct verbs (Fixed in #134), so, we do not need the biggest part of first commit :(

2) I want to support Java 6 as long as I can and use gradlew at the same time. Unfortunately, Gradle 2.14.1 is the latest version supports Java 6. So, I can't accept the third commit.

I will take your code for Self.java and apply it manually, thank you!

**Pull request closed without merging**